include/swift/AST/DiagnosticsClangImporter.def

```diff
@@ -258,8 +258,24 @@ ERROR(failed_base_method_call_synthesis,none,
       "failed to synthesize call to the base method %0 of type %0",
       (ValueDecl *, ValueDecl *))
 
-ERROR(both_returns_retained_returns_unretained,none,
-      "%0 cannot be annotated with both swift_attr('returns_retained') and swift_attr('returns_unretained') attributes", (const clang::NamedDecl*))
+ERROR(both_returns_retained_returns_unretained, none,
+      "%0 cannot be annotated with both swift_attr('returns_retained') and "
+      "swift_attr('returns_unretained') attributes",
+      (const clang::NamedDecl *))
+
+ERROR(returns_retained_or_returns_unretained_for_non_cxx_frt_values, none,
+      "%0 cannot be annotated with either swift_attr('returns_retained') or "
+      "swift_attr('returns_unretained') attribute because it is not returning "
+      "a 'SWIFT_SHARED_REFERENCE' type",
```

Comment:
nit: I don't think `SWIFT_SHARED_REFERENCE` needs to be in quotation marks here, since it is not a user-defined identifier. But then again, it seems like other diagnostic messages use quotation marks as well, so it may be better to stick with the existing convention.

Reply:
Yes good point! 
I think DiagnosticsClangImporter.def would need a refactoring. At least for diagnostics that we have recently added. I plan to do this in a follow up patch.

References: rdar://138810362, rdar://138812125, rdar://138805666, rdar://138874482,